wire/internal/wire/wire.go

```diff
@@ -172,9 +172,12 @@ func (g *gen) frame() []byte {
 		}
 		sort.Strings(imps)
 		for _, path := range imps {
-			// TODO(light): Omit the local package identifier if it matches
-			// the package name.
-			fmt.Fprintf(&buf, "\t%s %q\n", g.imports[path], path)
+			// Omit the local package identifier if it matches the package name.
+			if g.imports[path] == path {
```

Comment:
While this is commonly correct, the more reliable check is to get the package's identifier from the type checker and check those for equality. The information should be there, just another matter of plumbing it.

Reply:
PTAL.

Reply:
Not quite, `filepath.Base` is commonly correct, but not always. [`Package.Name()`](https://godoc.org/go/types#Package.Name) is what you should trust.